streaming/sender.c

```diff
@@ -13,24 +13,43 @@ void sender_start(struct sender_state *s) {
     buffer_flush(s->build);
 }
 
+static inline void rrdpush_sender_thread_close_socket(RRDHOST *host);
+
+#ifdef ENABLE_COMPRESSION
+/*
+* In case of stream compression buffer oveflow
+* Inform the user through the error log file and 
+* deactivate compression by downgrading the stream protocol.
+*/
+static inline void deactivate_compression(struct sender_state *s)
+{
+    error("STREAM_COMPRESSION: Deactivating compression to avoid stream corruption");
+    default_compression_enabled = 0;
+    s->rrdpush_compression = 0;
+    s->version = STREAM_VERSION_CLABELS;
```

Comment:
So currently CLABELS is 4 and compression is 5. What will happen with version 6 ?  It will support compression but when it fails it will fallback to 4 (losing the functionally added by version 6).

Reply:
I would prefer to merge this PR in the master and respect the stream versions (4 - clabels, 5 - compression, 6 - gap filling) in the gap filling PR.The stream version checks need to be gathered in a function because if we keep adding features in streaming the if/elif statements are going to increase the code complexity.